msal_extensions/token_cache.py

```diff
@@ -50,9 +50,21 @@ def modify(self, credential_type, old_entry, new_key_value_pairs=None):
             self._last_sync = time.time()
 
     def find(self, credential_type, **kwargs):  # pylint: disable=arguments-differ
-        with CrossPlatLock(self._lock_location):
-            self._reload_if_necessary()
-            return super(PersistedTokenCache, self).find(credential_type, **kwargs)
+        # Use optimistic locking rather than CrossPlatLock(self._lock_location)
+        retry = 3
+        for attempt in range(1, retry + 1):
+            try:
+                self._reload_if_necessary()
+            except Exception:  # pylint: disable=broad-except
+                # Presumably other processes are writing the file, causing dirty read
+                if attempt < retry:
+                    logger.debug("Unable to load token cache file in No. %d attempt", attempt)
+                    time.sleep(0.5)
+                else:
+                    raise  # End of retry. Re-raise the exception as-is.
+            else:  # If reload encountered no format error, the data is likely intact
```

Comment:
Wondering what is "no format error".

Reply:
Good point. Now it is reworded in the latest commit.